{"pageProps":{"questions":[{"id":"0KEE8ihvR3UHk6J5keAI","question_images":[],"choices":{"D":"instd, outstd. errstd are the names of pre-opened streams","A":"if invoking open () fails, an exception is raised","C":"open () is a function which returns an object that represents a physical file","B":"open () requires a second argument"},"answer_ET":"AC","exam_id":192,"timestamp":"2022-04-11 12:54:00","answers_community":["AC (100%)"],"question_id":16,"topic":"1","question_text":"Which of the following statements are true? (Choose two.)","url":"https://www.examtopics.com/discussions/python-institute/view/73512-exam-pcap-topic-1-question-112-discussion/","isMC":true,"answer_description":"","answer_images":[],"unix_timestamp":1649674440,"answer":"AC","discussion":[{"upvote_count":"6","timestamp":"1651666380.0","comment_id":"596781","content":"Selected Answer: AC\nanswer is correct: AC","poster":"macxsz"},{"comment_id":"987447","poster":"Ujjal_d","timestamp":"1692708600.0","content":"So it is A C right?","upvote_count":"1"},{"comment_id":"928868","upvote_count":"1","content":"A. If invoking open() fails, an exception is raised.\nD. instd, outstd, errstd are the names of pre-opened streams.","poster":"cufta05","timestamp":"1687302960.0"},{"comment_id":"584150","upvote_count":"1","content":"C is wrong. According to python doc \"mode\" is an optional string.\ninstd, outstd. errstd are preopened streams in sys module.\nWould go then for A and D.","comments":[{"timestamp":"1652538660.0","poster":"deckman","upvote_count":"5","content":"it's not instd, outstd, errstd... \"stdin\", \"stdout\" and \"stderr\" would be the correct ones","comment_id":"601632"},{"content":"it is stdin, stdout, and stderr","poster":"stuartz","timestamp":"1655746140.0","comment_id":"619398","upvote_count":"4"}],"timestamp":"1649674440.0","poster":"aserr"}]},{"id":"AfqM69As8T8PHxCIkHvW","question_id":17,"question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0008500001.png"],"answer":"D","choices":{"A":"the code is erroneous and it will not execute","B":"it outputs 1","C":"it outputs -1","D":"it outputs 0"},"discussion":[{"poster":"macxsz","upvote_count":"2","comment_id":"596784","timestamp":"1651666500.0","content":"Selected Answer: D\nD. it outputs 0"}],"topic":"1","answer_images":["https://www.examtopics.com/assets/media/exam-media/04235/0008500002.jpg"],"answers_community":["D (100%)"],"question_text":"What is the expected behavior of the following code?\n//IMG//","isMC":true,"answer_ET":"D","exam_id":192,"timestamp":"2022-05-04 14:15:00","answer_description":"","unix_timestamp":1651666500,"url":"https://www.examtopics.com/discussions/python-institute/view/75151-exam-pcap-topic-1-question-113-discussion/"},{"id":"fCvjkcWZk855qHyXUnat","answer_description":"","question_text":"Which of the following snippets will execute without raising any unhandled exceptions? (Choose two.)\nA.\n//IMG//\n\nB.\n//IMG//\n\nC.\n//IMG//\n\nD.\n//IMG//","question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0008600001.png","https://www.examtopics.com/assets/media/exam-media/04235/0008600002.png","https://www.examtopics.com/assets/media/exam-media/04235/0008700001.png","https://www.examtopics.com/assets/media/exam-media/04235/0008700002.png"],"discussion":[{"poster":"fermins","upvote_count":"3","content":"try:\n print(float('1e1'))\nexcept (ValueError, NameError):\n print(float('1a1'))\nelse:\n print(float('101'))\n\nRises no exception. Both A and D are correct","comment_id":"1138331","timestamp":"1706864280.0"},{"timestamp":"1692252660.0","comment_id":"983276","upvote_count":"3","poster":"emanuelcar990","content":"try:\n print(float(\"1e1\"))\nexcept (ValueError, NameError):\n print(float(\"1a1\"))\nelse:\n print(float(\"101\"))\ntry:\n print(0/1)\nexcept:\n print(1/1)\nelse:\n print(2/1)\n\nare the correct , because float(\"1e1\") = 10,0 then the else : float(\"101\") = 101.0\nthe other code =\n0 /1 = 1 then else: 2/1 = 2\nso A D"},{"content":"A and D are correct.","comment_id":"741734","timestamp":"1670764800.0","upvote_count":"4","poster":"Nenggg"},{"poster":"Dav023","upvote_count":"3","content":"A and D are correct.","comment_id":"682746","timestamp":"1664459460.0"},{"upvote_count":"3","comment_id":"619400","content":"A and D are correct. B fails, C has no exception to handle","poster":"stuartz","timestamp":"1655746500.0"},{"timestamp":"1653982680.0","poster":"Baldridge","content":"A D are correct","comment_id":"609602","upvote_count":"3"},{"comment_id":"594969","timestamp":"1651316760.0","content":"Only D is right.\nA raises another exception after the first ValueError exception","upvote_count":"2","poster":"macxsz"},{"comment_id":"591773","upvote_count":"3","content":"AD should be right","timestamp":"1650897420.0","poster":"MTLE"}],"answer_ET":"D","isMC":false,"timestamp":"2022-04-25 16:37:00","answer_images":["https://www.examtopics.com/assets/media/exam-media/04235/0008800001.jpg"],"topic":"1","unix_timestamp":1650897420,"answer":"D","question_id":18,"exam_id":192,"answers_community":[],"url":"https://www.examtopics.com/discussions/python-institute/view/74485-exam-pcap-topic-1-question-114-discussion/"},{"id":"PA6N63ByLsXusrvfnZT3","exam_id":192,"answer_images":[],"topic":"1","answer_ET":"BD","isMC":true,"timestamp":"2022-04-25 16:39:00","question_id":19,"question_text":"Which of the following expressions evaluate to True? (Choose two.)","question_images":[],"choices":{"A":"'8' + '8' !=2 * '8'","D":"121 + 1 ==int('1' + 2 * '2')","B":"'xYz'.lower() > 'XY'","C":"float('3.14') == str('3.' + '14')"},"unix_timestamp":1650897540,"url":"https://www.examtopics.com/discussions/python-institute/view/74486-exam-pcap-topic-1-question-115-discussion/","answer":"BD","answer_description":"","answers_community":["BD (100%)"],"discussion":[{"comment_id":"1138335","poster":"fermins","timestamp":"1706864520.0","upvote_count":"1","content":"Selected Answer: BD\nprint('xYz'.lower() > 'XY') => True\nprint(121 + 1 ==int('1' + 2 * '2')) => True"},{"upvote_count":"3","comment_id":"835056","timestamp":"1678455840.0","poster":"dicksonpwc","content":"answer should be B, D\nprint('8' + '8' !=2*'8') # ans A, False\nprint('xYz'.lower() > 'XY') # ans B, True\nprint(float('3.14')==str('3.''14')) # ans C, False\nprint(121+1 == int('1'+2*'2')) # ans D, True"},{"content":"Selected Answer: BD\nshould be B & D","timestamp":"1677630000.0","upvote_count":"3","poster":"ivanbicalho","comment_id":"825402"},{"poster":"Mallie","comment_id":"751317","timestamp":"1671559680.0","content":"Selected Answer: BD\nprint('xYz'.lower() > 'XY')\nTrue\nB & D","upvote_count":"1"},{"timestamp":"1670764380.0","content":"Selected Answer: BD\nprint('8' + '8') -> 88\nprint(2 * '8') -> 88\nso A is False","poster":"Nenggg","comment_id":"741725","upvote_count":"1"},{"poster":"haly","upvote_count":"1","timestamp":"1665213300.0","comment_id":"689093","content":"Selected Answer: BD\ncorrect answers"},{"timestamp":"1664459760.0","content":"Selected Answer: BD\nright ones","comment_id":"682751","upvote_count":"1","poster":"Dav023"},{"comment_id":"604177","upvote_count":"2","poster":"Dr_Alzuhairy21","content":"Selected Answer: BD\nThe correct answer is BD","timestamp":"1653004320.0"},{"upvote_count":"2","timestamp":"1651274880.0","content":"B and D","comment_id":"594724","poster":"macxsz"},{"upvote_count":"1","poster":"MTLE","comment_id":"591775","timestamp":"1650897540.0","content":"BD - A would be true if we had =="}]},{"id":"OVtb5XLGpsS1y4iJfnSw","isMC":true,"answer_ET":"A","answer_images":[],"answers_community":["A (100%)"],"discussion":[{"poster":"zantrz","upvote_count":"1","comment_id":"1148295","content":"Selected Answer: A\nif we magle correctly, we would get the output 2, which is not in the answer list. \nThe first call to o._Class__foo() increments Var by 1 and returns the updated value, which is 1.\nThe second call to o._Class__foo() again increments Var by 1 (now it's 2) and returns the updated value, which is 2.","timestamp":"1707753060.0"},{"comment_id":"1144496","upvote_count":"1","content":"Mangling error","poster":"Damon54","timestamp":"1707399840.0"},{"timestamp":"1671971040.0","content":"Selected Answer: A\nUsing correct mangling, o._Class__foo(), gives an output of 2.\nSo answer is A.","comment_id":"755627","upvote_count":"2","poster":"Mallie"},{"timestamp":"1651666980.0","upvote_count":"2","comments":[{"poster":"Iamrandom","timestamp":"1704277500.0","upvote_count":"1","comment_id":"1112655","content":"which typos?"}],"comment_id":"596789","content":"Selected Answer: A\nbecause of the typos:\nA. it raises an exception","poster":"macxsz"}],"url":"https://www.examtopics.com/discussions/python-institute/view/75152-exam-pcap-topic-1-question-116-discussion/","answer_description":"","exam_id":192,"question_text":"What is the expected behavior of the following code?\n//IMG//","unix_timestamp":1651666980,"timestamp":"2022-05-04 14:23:00","choices":{"B":"it outputs 3","D":"it outputs 6","C":"it outputs 1","A":"it raises an exception"},"question_id":20,"answer":"A","question_images":["https://www.examtopics.com/assets/media/exam-media/04235/0008900001.png"],"topic":"1"}],"exam":{"name":"PCAP","lastUpdated":"12 Apr 2025","numberOfQuestions":141,"isBeta":false,"isImplemented":true,"id":192,"isMCOnly":false,"provider":"Python-institute"},"currentPage":4},"__N_SSP":true}